pkgs/applications/networking/firewalld/default.nix

```diff
@@ -0,0 +1,110 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, docbook_xml_dtd_42
+, docbook_xsl
+, glib
+, gobject-introspection
+, gtk3
+, intltool
+, libnotify
+, libxml2
+, libxslt
+, networkmanager-applet
+, pkg-config
+, python3
+, wrapGAppsHook
+, withGui ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "firewalld";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "firewalld";
+    repo = "firewalld";
+    rev = "v${version}";
+    sha256 = "sha256-w8TbovIhOhJAUZWbKdBd/+db8Hro/ttlxWZDcrCXX4Q=";
+  };
+
+  postPatch = ''
+    for file in doc/xml/transform-{html,man}.xsl.in; do
+      substituteInPlace $file \
+        --replace "http://docbook.sourceforge.net/release/xsl/current" "${docbook_xsl}/xml/xsl/docbook"
+    done
+
+    for f in doc/xml/*.xml* ; do
+      substituteInPlace $f \
+        --replace "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd" "${docbook_xml_dtd_42}/xml/dtd/docbook/docbookx.dtd"
+    done
+
+    substituteInPlace src/firewall/config/__init__.py.in \
+      --replace "/usr/share" "$out/share"
+
+    for file in config/firewall-{applet,config}.desktop.in; do
+      substituteInPlace $file \
+        --replace "/usr/bin/" "$out/bin/"
+    done
+
+    patchShebangs --host src/gtk3_chooserbutton.py src/gtk3_niceexpander.py src/tests/python/
+  '' + lib.optionalString withGui ''
+    substituteInPlace src/firewall-applet.in \
+      --replace "/usr/bin/nm-connection-editor" "${networkmanager-applet}/bin/nm-conenction-editor"
+  '';
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    autoreconfHook
+    glib
+    intltool
+    libxml2
+    libxslt
+    pkg-config
+    python3
+    python3.pkgs.wrapPython
+  ] ++ lib.optionals withGui [
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    glib
+  ] ++ lib.optionals withGui [
+    gobject-introspection
```

Comment:
This is just for setup hook, right? It should likely go to `nativeBuildInputs` but `strictDeps` prevent it: https://github.com/NixOS/nixpkgs/issues/56943